webapp/src/pages/Settings.tsx

```diff
@@ -41,38 +69,434 @@ const Settings: React.FC = () => {
     />
   );
 
+  const displayConfigTitle = (title: string, link?: string) => (
+    <Box display="flex" gap={1}>
+      <Typography variant="body1">{title}</Typography>
+      {link && (
+        <Link
+          href={DOCS_URL + link}
+          variant="body2"
+          color="secondary"
+          target="_blank"
+        >
+          <Info color="primary" fontSize="small" />
+        </Link>
+      )}
+    </Box>
+  );
+
+  const displayReadonlyFields = (
+    label: React.ReactElement,
+    defaultValue: string | undefined
+  ) =>
+    defaultValue ? (
+      <TextField
+        sx={{ width: FIELDS_WIDTH }}
+        label={label}
+        defaultValue={defaultValue}
+        InputProps={{
+          readOnly: true,
+        }}
+        variant="standard"
+      />
+    ) : (
+      <></>
+    );
+
+  const displaySubFields = (
+    config: keyof AzimuthConfig,
+    label: string,
+    field: string,
+    limit: InputBaseComponentProps,
+    value: number | undefined
+  ) => (
+    <TextField
+      sx={{ width: FIELDS_WIDTH, marginLeft: 5, marginBottom: 2 }}
+      id={field}
+      label={label}
+      type="number"
+      InputLabelProps={{
+        shrink: true,
+      }}
+      defaultValue={value}
```

Comment:
We should always "control" our inputs, and never use `defaultValue`. Otherwise, if a field is affected by any other input, like a reset button, a relead button, or even the URL, then it would not work.
```suggestion
      value={value}
```

Reply:
Don't forget this two-week-old comment.

Reply:
I have already taken care of this commit, But unfortunately, the field behaves like read only if I change to value attribute. I am not able to find the reason why. I am figuring out as well why it is behaving this way. I could also see that you used text field in class overlap component. have you faced this issue earlier?

Reply:
In the future, if you try to apply a review suggestion and you get an issue, please reply (in less than 2 weeks) so we can look at the issue together.
At a glance, I have no idea what issue you faced. Let's Zoom tomorrow to try it together?